### ISSUE TITLE
Pkl support

### ISSUE THREAD

**Issue by leonardwoo** [FasterXML/jackson-dataformats-text#464]
Apple configuration as code language with rich validation and tooling.

https://github.com/apple/pkl

**Comment by cowtowncoder** [comment]
Looks a bit like YAML, but also is a ... programming language? Might be challenging to expose via Jackson Streaming API but who knows. If anyone wants to take a crack, would be happy to help with details!

**Comment by yawkat** [comment]
i think you need to evaluate the whole file at a time, but with config formats that's not too bad. probably will have to do it the same way as the toml module.

**Comment by cowtowncoder** [comment]
Yeah agreed, cannot do truly incremental processing and no, probably not a big deal for config formats. But the complexity of evaluating expressions may be challenging.
